src/utils/stac_rest.py

```diff
@@ -1,19 +1,26 @@
 import requests
 
+from utils.logging_config import logger
+
 
 def post_or_put(url: str, data: dict):
     """
     Post or put data to url
     """
+    try:
+        response = requests.post(url, json=data)
+
+        if response.status_code == 409:
+            response = requests.put(url, json=data)
 
-    response = requests.post(url, json=data)
-    if response.status_code == 409:
-        response = requests.put(url, json=data)
-        if not response.status_code == 404:
-            response.raise_for_status()
-    else:
         response.raise_for_status()
 
+    except requests.exceptions.RequestException as e:
+        logger.error(f"Error during post or put: {e}")
+        return None
```

Comment:
I don't think it's a good idea to return None here. Why did you di this instead or raising the error?

Reply:
I left it for the moment to do the tests, I forgot to make the adjustment